framework/util/page_guard_manager.h

```diff
@@ -225,9 +235,18 @@ class PageGuardManager
     const bool               enable_copy_on_map_;
     const bool               enable_separate_read_;
     const bool               unblock_sigsegv_;
+    bool                     signal_handler_watcher_;
 
     // Only applies to WIN32 builds and Linux/Android builds with PAGE_GUARD_ENABLE_UCONTEXT_WRITE_DETECTION defined.
     const bool enable_read_write_same_page_;
+
+#if !defined(WIN32)
+    pthread_t    signal_handler_watcher_thread;
+    static void* signal_handler_watcher_f(void*);
+
+    // Signal watcher thread period in miliseconds
+    static constexpr int64_t signal_watcher_thread_period{ 100 };
```

Comment:
Should we default to 0ms (AKA spin)? 100ms is enough time to lose a whole frame. There is always going to be a race that generates rare failures, even if the monitor thread spins, but maybe minimising the duration by default is the best we can do?

Reply:
We had a similar discussion with Brad concerning the frequency the check is performed.
I guess it depends on how you look at it. The way I see it, for something that will only happen once or twice and most probably at init time, perhaps 100ms is too frequent. Fortnite seems to work with 100ms. We can make this configurable by env variables